res/schema/audit-layout.sql

```diff
@@ -43,6 +43,24 @@ CREATE TABLE IF NOT EXISTS message (
 );
 
 
+CREATE TABLE IF NOT EXISTS metadata (
+  id                INTEGER PRIMARY KEY NOT NULL,
+  context           VARCHAR(36)         NOT NULL REFERENCES execution (context),
+  kind              VARCHAR(256)        NOT NULL DEFAULT '',
+  name              VARCHAR(1024)       NOT NULL DEFAULT '',
+  metadata          TEXT                         DEFAULT NULL,
```

Comment:
Seeing this and also the dialog data, I think you might want to take an inspiration here on the message_data and deduplicate it.
You would just store a hash of the data here, as this will be stored multiple times and every execution adds the same data here every time.